deltachat-jsonrpc/src/api/types/events.rs

```diff
@@ -138,21 +171,57 @@ pub enum EventType {
     /// A single message is sent successfully. State changed from  DC_STATE_OUT_PENDING to
     /// DC_STATE_OUT_DELIVERED, see `Message.state`.
     #[serde(rename_all = "camelCase")]
-    MsgDelivered { chat_id: u32, msg_id: u32 },
+    MsgDelivered {
+        /// ID of the chat which the message belongs to.
+        chat_id: u32,
+
+        /// ID of the message that was successfully sent.
+        msg_id: u32,
+    },
 
     /// A single message could not be sent. State changed from DC_STATE_OUT_PENDING or DC_STATE_OUT_DELIVERED to
     /// DC_STATE_OUT_FAILED, see `Message.state`.
     #[serde(rename_all = "camelCase")]
-    MsgFailed { chat_id: u32, msg_id: u32 },
+    MsgFailed {
+        /// ID of the chat which the message belongs to.
+        chat_id: u32,
+
+        /// ID of the message that could not be sent.
+        msg_id: u32,
+    },
 
     /// A single message is read by the receiver. State changed from DC_STATE_OUT_DELIVERED to
     /// DC_STATE_OUT_MDN_RCVD, see `Message.state`.
     #[serde(rename_all = "camelCase")]
-    MsgRead { chat_id: u32, msg_id: u32 },
+    MsgRead {
+        /// ID of the chat which the message belongs to.
+        chat_id: u32,
+
+        /// ID of the message that was read.
+        msg_id: u32,
+    },
 
-    /// A single message is deleted.
+    /// A single message was deleted.
+    ///
+    /// This event means that the message will no longer appear in the messagelist.
+    /// UI should remove the message from the messagelist
+    /// in response to this event if the message is currently displayed.
+    ///
+    /// The message may have been explicitly deleted by the user or expired.
+    /// Internally the message may have been removed from the database,
+    /// moved to the trash chat or hidden.
+    ///
+    /// This event does not indicate the message
+    /// deletion from the server.
     #[serde(rename_all = "camelCase")]
-    MsgDeleted { chat_id: u32, msg_id: u32 },
+    MsgDeleted {
+        /// ID of the chat where the message was prior to deletion.
+        /// Never 0 or trash chat.
```

Comment:
```suggestion
        /// Never 0.
```

> The trash chat is just an internal implementation detail of core, to the UIs a trashed message doesn't exist, so I don't think it's not necessary to mention it here.

Does this work for you?

Reply:
> Does this work for you?

Does what work for me?

Reply:
The suggestion, to make the line read `/// Never 0.`.